### PR TITLE
REST API Framework

### PR DESCRIPTION
**Pull request by Juan-Garcia96**
REST endpoints for networks and devices as well as GUI progress

### CODE REVIEW

**Comment by Juan-Garcia96** [comment]
Will need to add DatabaseConnectionError exception in to this PR as well since I did not add in the last commit

File: src/chariot/device/DeviceAdapterFactory.py [hunk 1]
```diff
@@ -30,10 +32,40 @@ def getsupportedDevices(self) -> JSONDict:
 
     def getDeviceInformation(self, deviceName: str) -> JSONDict:
         if deviceName not in self._supportedDevices:
-            # raise UnsupportedDeviceError
-            raise AssertionError
+            raise DeviceNotSupported(ErrorStrings.ERR_Device_Not_Supported)
```
Comment (kac486) [inline src/chariot/device/DeviceAdapterFactory.py]: indicate bad device that is not supported (deviceName)

File: src/chariot/api/webserver.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    payloadConfig = request.get_json()
+
+    dbConfig: DatabaseConfiguration = DatabaseConfigurationFactory.getInstance(payloadConfig)
+
+    # with configuration validated, now use the factory to create a dbWriter instance
+    dbWriter: DatabaseWriter = DatabaseWriterFactory.getInstance(dbConfig)
+
+    try:
+        dbWriter.connect()
+    except:
```
Comment (kac486) [inline src/chariot/api/webserver.py]: use "except e" to get a more descriptive error message

File: src/chariot/utility/PayloadParser.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @staticmethod
+    def getNetworkDescription(requestContent) -> str:
+        # first ensure that a network name has been given to specify which network is to be modified
+        networkName: str = requestContent.get('Description')
+        return networkName
+
+    @staticmethod
+    def getNewNetworkName(requestContent) -> str:
+        # first ensure that a network name has been given to specify which network is to be modified
+        newNetworkName: str = requestContent.get('NewName')
```
Comment (kac486) [inline src/chariot/utility/PayloadParser.py]: be more consistent with naming "NewName" vs "deviceId"

**Pull request merged**